src/mongo/delegates/Updates.class.php

```diff
@@ -144,6 +151,12 @@ public function saveChanges(
         $context=null,
         $description=null)
     {
+        $this->applyPreHooks($this->saveChangesHooks,array(
```

Comment:
I don't have a better alternative, offhand, but this feels kind of awkward.

I like the intention, though.  I kind of wish there was an easier way to register these hooks, but no matter what, I don't know of an elegant way to pass local variables to the event hook without doing it like this.